Fixes stopword requirement

p-galligan opened the pull request:
Removes the `~1` from the end of `processed` in search.js so it actually matches against stopwords. Fixes #92 

Should be noted that it does make searches a little less exact when included because it's adding that word as an `OR` search, but I'm not sure the best solution for that.

Commit:
fix stopword check

Removes the `~1` from the end of `processed` in search.js so it actually matches against stopwords.

helrond reviewed (commented):
This will definitely work. I'm wondering, though, for readabilty's sake, if we don't want to do something like:
```
function preProcessQueryTerm(term) {
    const charsRemoved = term.replace(/:|"|'|~|\^/g, "")
    const fuzzyDistance = 1
    const fuzzyTerm = charsRemoved.concat(`~${fuzzyDistance}`)
    if (!charsRemoved) {
      return null
    }
    else {
      const targetedTerm = searchField.length ? `${searchField}:${fuzzyTerm}` : fuzzyTerm
      return stopWords.includes(charsRemoved) ? `${targetedTerm}` : `+${targetedTerm}`
    }
  }
```
It's more code, but it makes what's happening a bit more explicit, and the variable names are a bit more meaningful...

I haven't tested this code.

Commit:
Rework preProcessQueryTerm

Reworks preprocessing function.

Fuzzy distance is a settable variable and separates character replacement and fuzzy distance appending.

p-galligan commented:
Ok, based on some general testing, this does work. This makes sense and works with stop words.

Other search functionality with special character searches and wildcard searches also appear to work as well.

helrond reviewed (approved):
Looks good!
